Vehicle.h

```diff
@@ -3,26 +3,23 @@
 #include <string>
 #include <iostream>
 
+struct HighwayPatrol;
+struct Highway;
+
 struct Vehicle
 {
-    Vehicle(const std::string& n) : name(n) { }
-    
-    virtual ~Vehicle() = default;
-    Vehicle(const Vehicle&) = default;
-    Vehicle& operator=(const Vehicle&) = default;
+    Vehicle(const std::string& n);
     
-    virtual void setSpeed(int s)
-    {
-        std::cout << "\nHighway is adjusting " << name << "'s speed to " << s << " mph" << std::endl;
-        speed = s;
-    }
+    virtual ~Vehicle();
+    Vehicle(const Vehicle&);
+    Vehicle& operator=(const Vehicle&);
     
-    virtual void tryToEvade()
-    {
-        std::cout << name << ": you'll never take me alive, ya dirty coppers!" << std::endl;
-        setSpeed(100);
-    }
+    virtual void setSpeed(int s);
+    virtual void tryToEvade();
+
 protected:
     int speed = 0;
     std::string name;
+    friend Highway;
```

Comment:
I don't think `Highway` needs access to Vehicle's private member variables.  
- [ ] try removing `friend Highway` and see if it explodes or if it's fine.

Reply:
It did not explode 🚫 💥, fixed!